### ISSUE TITLE
[Question] Titanic tutorial, neural network parameters

### ISSUE THREAD

**Issue by shazz** [tflearn/tflearn#396]
Hi,

Probably a dumb question but... in the tutorial, when the neural network is built, why 32 in `net = tflearn.fully_connected(net, 32)` ?

42 I could understand but 32... I don't :)

Thanks !

**Comment by OlivierData** [comment]
32 is the number of units for this layer.
You should read [Feedforward neural network](https://en.wikipedia.org/wiki/Feedforward_neural_network).

**Comment by shazz** [comment]
Ok, I was not clear, I meant how did you decide to set 32 units for this layer in this example, I did some tests using only one hidden layer and 10 units for example and the model accuracy is as good.

Is it just random ?

**Comment by OlivierData** [comment]
It's not really random but you are right, a wide network is not always the best solution and it is computationally intensive. As you said you can reach a good accuracy with a small network.

Actually you have to find out what is the best number of units for your model. If you want to know more about that, you should seek for "deep learning hyperparameter optimization".

**Comment by shazz** [comment]
Thanks !
